test/controllers/order_items_controller_test.rb

```diff
@@ -1,9 +1,135 @@
 require "test_helper"
 
 describe OrderItemsController do
-  # it "does a thing" do
-  #   value(1+1).must_equal 2
-  # end
-  
-end
+  describe 'guest users' do
+    describe "create" do
+      before do
+        @user = User.create(username: "georgina", email: "geor@gmail.com")
+        @product_cactus = Product.create(user_id: @user.id, name: "cactus", description: "cool product", price: 1.9, photo_url: "url", stock: 3)
+        @product_flower = Product.create(user_id: @user.id, name: "flower", description: "cool flower", price: 1.5, photo_url: "url", stock: 3)
+        @order = Order.create(address: "x", name: "x", cc_num: "x", cvv_code: "x", zip: "x", email: "blank@blank.com")
+        @order_item_hash = {
+          order_item: {
+            product_id: @product_cactus.id,
+            quantity: 3,
+            order_id: @order.id,
+          }
+        }
+      end
+
+      it 'creates a new order_item successfully with valid data while not logged in, and redirects the user to the products page' do
+        expect {
+          post order_items_path, params: @order_item_hash[:order_item]
+        }.must_differ 'OrderItem.count', 1
+        
+        must_redirect_to products_path
+      end
+    end
+
+    describe 'delete' do
+      before do
+        @user = User.create(username: "georgina", email: "geor@gmail.com")
+        @product_cactus = Product.create(user_id: @user.id, name: "cactus", description: "cool product", price: 1.9, photo_url: "url", stock: 3)
+        @product_flower = Product.create(user_id: @user.id, name: "flower", description: "cool flower", price: 1.5, photo_url: "url", stock: 3)
+        @order = Order.create(address: "x", name: "x", cc_num: "x", cvv_code: "x", zip: "x", email: "blank@blank.com")
```

Comment:
I've been thinking about this more. We should ask the team about the order validation. They way we have to put "x's in there to start is no different than if we don't have requirements on those fields. If we make them not required to start, we could still make the corresponding form fields required at checkout to ensure that we DO capture that data.

Reply:
I agree! Let's discuss this on Monday